src/lib/input/input-wrapper.ts

```diff
@@ -0,0 +1,289 @@
+  get _inputPlaceholder(): string { return this._getterSetterOnlyInputPlaceholder; }
+  set _inputPlaceholder(value: string) {
+    this._getterSetterOnlyInputPlaceholder = value;
+    this._validatePlaceholders();
+  }
+  private _getterSetterOnlyInputPlaceholder = '';
+
+  /** The type attribute of the `input` (or "text" if element is a `textarea`). */
+  private get _inputType(): string { return this._getterSetterOnlyInputType; }
+  private set _inputType(value: string) {
+    this._getterSetterOnlyInputType = value || 'text';
+    this._validateInputType();
+  }
+  private _getterSetterOnlyInputType = 'text';
+
+  /** The value of the `input` or `textarea`. */
+  private _inputValue = '';
+
+  /** The native `input` or `textarea` element. */
+  private _inputElement: HTMLInputElement | HTMLTextAreaElement;
+
+  /** A `MutationObserver` to observe the `input` or `textarea`. */
+  private _inputObserver = new MutationObserver(mutations => {
+    for (let mutation of mutations) {
+      switch (mutation.attributeName) {
+        case 'disabled':
+          this._inputDisabled = this._inputElement.disabled;
+          break;
+        case 'id':
+          this._inputId = this._inputElement.id;
+          break;
+        case 'type':
+          // We need to use getAttribute since `type="date"` is not supported on Firefox and returns
+          // `"text"` if we just do `inputElement.type`.
+          this._inputType = this._inputElement.getAttribute('type');
+          break;
+        case 'placeholder':
+          this._inputPlaceholder = this._inputElement.placeholder;
+          break;
+        case 'required':
+          this._inputRequired = this._inputElement.required;
+          break;
+      }
+    }
+  });
+
+  /** A map of event listeners to install on the `input` or `textarea`. */
+  private _inputListeners: { [key: string]: () => void } = {
+    'blur': () => { this._focused = false; },
+    'focus': () => { this._focused = true; },
+    'input': () => { this._inputValue = this._inputElement.value; }
+  };
+
+  constructor(private _elementRef: ElementRef) {}
+
+  ngAfterContentInit() {
+    this._initInputEl();
+    this._validateHints();
+
+    // Trigger validation when the hint children change.
+    this._hintChildren.changes.subscribe(() => {
+      this._validateHints();
+    });
+  }
+
+  ngOnChanges(changes: {[key: string]: SimpleChange}) {
+    if (changes['hintLabel']) {
+      this._validateHints();
+    }
+  }
+
+  /** Set focus on the input element. */
+  _focusInput() {
+    if (this._inputElement) {
+      this._inputElement.focus();
+    }
+  }
+
+  /** Whether the input has a placeholder. */
+  _hasPlaceholder(): boolean {
+    return !!this._inputPlaceholder || !!this._placeholderChild;
+  }
+
+  /** Initialize the `input` or `textarea` element. */
+  private _initInputEl() {
+    // Find input or textarea element.
+    let inputEls = this._elementRef.nativeElement.querySelectorAll('input, textarea');
```

Comment:
@jelbourn Refresh my memory that we cannot use `@ContentChild()`? I thought we could but might have been mistaken.